### ISSUE TITLE
CLIEngine should not require the `ignorePath` file to exist

### ISSUE THREAD

**Issue by saadq** [eslint/eslint#10248]
**Tell us about your environment**

* **ESLint Version:** v4.19.1
* **Node Version:** 5.8.0
* **npm Version:** v9.11.1

**What parser (default, Babel-ESLint, etc.) are you using?**
Default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
const cli = new CLIEngine({
  parserOptions: {
    ecmaVersion: 8
  },
  useEslintrc: false,
  ignorePath: '.linterignore',
  rules: {
    'prefer-const': 2
  }
})
```
</details>

<br />

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I have a minimal repo that reproduces the issue [here](https://github.com/saadq/misc/tree/master/eslint-ignore-path). But essentially, I have this code: 

`index.js`
```js
const { CLIEngine } = require('eslint')

const cli = new CLIEngine({
  useEslintrc: false,
  ignorePath: '.linterignore',
  parserOptions: {
    ecmaVersion: 8
  },
  rules: {
    'prefer-const': 2
  }
})

const report = cli.executeOnFiles(['test.js'])
const formatter = cli.getFormatter('stylish')

console.log(formatter(report.results))
```

Which I run by doing `node index.js`.

**What did you expect to happen?**
It should use the `.linterignore` file if it exists, but if the file doesn't exist there shouldn't be any problem.

**What actually happened? Please include the actual, raw output from ESLint.**
If the `.linterignore` file doesn't exist, it complains about a missing ignored file:

```
> eslint-ignore-path@0.1.0 test /Users/squadri/Projects/misc/eslint-ignore-path
> node index.js

/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/eslint/lib/ignored-paths.js:146
                    throw e;
                    ^

Error: Cannot read ignore file: .linterignore
Error: ENOENT: no such file or directory, stat '.linterignore'
    at Object.fs.statSync (fs.js:965:11)
    at new IgnoredPaths (/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/eslint/lib/ignored-paths.js:142:24)
    at lodash.memoize.optionsObj (/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/eslint/lib/util/glob-util.js:115:13)
    at memoized (/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/lodash/lodash.js:10543:27)
    at globPatterns.forEach.pattern (/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/eslint/lib/util/glob-util.js:162:34)
    at Array.forEach (<anonymous>)
    at Object.listFilesToProcess (/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/eslint/lib/util/glob-util.js:158:18)
    at CLIEngine.executeOnFiles (/Users/squadri/Projects/misc/eslint-ignore-path/node_modules/eslint/lib/cli-engine.js:518:35)
    at Object.<anonymous> (/Users/squadri/Projects/misc/eslint-ignore-path/index.js:14:20)
    at Module._compile (internal/modules/cjs/loader.js:654:30)
npm ERR! Test failed.  See above for more details.
```

**Comment by platinumazure** [comment]
Hi @saadq, thanks for the issue.

I think this is working as designed. The value of throwing an exception if the file doesn't exist is to help catch typos so that users can be sure the ignore file is being read. This has more implications than just linting: ESLint does spend a lot of time searching for files, and entire directories can be skipped if they're configured correctly in an ignore file, saving a lot of processing time.

Can you explain your use case? Why do you need ignore file existence checks to fail silently? Maybe your use case is more compelling than what I presented above, but I don't know.

**Comment by saadq** [comment]
Well for my use case, I was working on a tool that would act as a zero-config wrapper around ESLint (or TSLint if `--typescript` is passed). I just wanted to add the same functionality as `.eslintignore` to it, so that if a user has an ignore file it will use those, and if not then there's no issue.

From what I understand, with the current state of things the user would not be able to use this tool unless they had a `.linterignore` file in their directory which seems problematic if the user doesn't need it.

**Comment by platinumazure** [comment]
If your tool blindly passes ".linterignore" to the CLIEngine constructor,
yes, that is the case. That said, there's nothing stopping you from
checking it in your tool:

```js
const fs = require("fs");

const cliEngine = new CLIEngine ({
    ignorePath: fs.fileExistsSync(".linterignore") ? ".linterignore" : null
});
```

**Comment by saadq** [comment]
Yes, I suppose that's fair. I guess I'll close this out then if this is by design, thanks for the quick replies :)
